### PR TITLE
Update web3j to be compatible with new private transaction receipt

### PR DESCRIPTION
**Pull request by josh-richardson**
### What does this PR do?
Adds some fields to web3j private transaction receipts

### Why is it needed?
Besu will soon have a more extensive private transaction receipt.

### CODE REVIEW

**Comment by josh-richardson** [comment]
@iikirilov This is ready for review.

File: besu/src/test/java/org/web3j/protocol/besu/ResponseTest.java [hunk 1]
```diff
@@ -312,7 +313,9 @@ public void testPrivGetTransactionReceipt() {
                                         "0x0000000000000000000000000000000000000000000000000000000000000000",
                                         "mined",
                                         Collections.singletonList(
-                                                "0x59ebeb90bc63057b6515673c3ecf9438e5058bca0f92585014eced636878c9a5"))));
+                                                "0x59ebeb90bc63057b6515673c3ecf9438e5058bca0f92585014eced636878c9a5"))),
```
Comment (iikirilov) [inline besu/src/test/java/org/web3j/protocol/besu/ResponseTest.java]: nit: extra line?

File: core/src/main/java/org/web3j/tx/Contract.java [hunk 2]
```diff
@@ -372,9 +372,11 @@ TransactionReceipt executeTransaction(
                     String.format(
                             "Transaction has failed with status: %s. "
                                     + "Gas used: %d. (not-enough gas?)",
-                            receipt.getStatus(), receipt.getGasUsed()));
+                            receipt.getStatus(),
+                            receipt.getGasUsedRaw() != null
+                                    ? receipt.getGasUsed()
+                                    : BigInteger.valueOf(0)));
```
Comment (iikirilov) [inline core/src/main/java/org/web3j/tx/Contract.java]: this should be implemented differently

File: besu/src/test/java/org/web3j/protocol/besu/ResponseTest.java (hunk 1 above)
Comment (josh-richardson) [reply besu/src/test/java/org/web3j/protocol/besu/ResponseTest.java]: Not that I can see

File: core/src/main/java/org/web3j/tx/Contract.java (hunk 2 above)
Comment (iikirilov) [reply core/src/main/java/org/web3j/tx/Contract.java]: can you do something like "Gas used: unknown" if it is not available

Comment (iikirilov) [reply core/src/main/java/org/web3j/tx/Contract.java]: It would also be useful to log the revert reason here if it is available

Comment (josh-richardson) [reply core/src/main/java/org/web3j/tx/Contract.java]: Updated this to return unknown gas used, showing a revert reason would introduce a cyclic dependency.

**Pull request merged**